libbeat/outputs/fileout/config.go

```diff
@@ -14,3 +20,16 @@ var (
 		RotateEveryKb: 10 * 1024,
 	}
 )
+
+func (c *config) Validate() error {
+	if c.Filename == "" && c.Index == "" {
```

Comment:
I mostly copied in-code validations into Validate methods. If filename is missing, the file outputer will fall back to index. I think it's fine, given c.Index equals the beatname by default. E.g. for packetbeat the filebane will be packetbeat.log if not explicitly configured.